astrocut/cube_cut.py

```diff
@@ -729,10 +729,10 @@ def _build_tpf(self, cube_fits, img_cube, uncert_cube, cutout_wcs_dict, aperture
 
         # Adding flux and flux_err (data we actually have!)
         pixel_unit = 'e-/s' if self.product == 'SPOC' else 'e-'
+        flux_err_array = uncert_cube if self.product == 'SPOC' else empty_arr
```

Comment:
i'm a little uncertain about this. numpy array assignment is done by reference. i'm just worried that because we are using empty_arry here to assign into cadence_arry, or appending to columns, that something might change inside of `empty_arr` and then would be applied to flux_err_arry at this point.

e.g. to illustrate what i'm worried about

```python
>>> import numpy as np
>>> empty_arr = np.zeros((2,3))
>>> other_arr = empty_arr[:,1]
>>> other_arr[0] = 1
>>> other_arr
array([1., 0.])
>>> empty_arr
array([[0., 1., 0.],
       [0., 0., 0.]])
```

maybe on this line, we should assign flux_err_arry to just `np.zeros(img_cube.shape)` to be sure it's empty?

maybe we're not changing `cadence_array` so this isn't an issue?

Reply:
another think i'm wondering -- should the error be zero? or should it be np.NaN?

Reply:
I thought about this too.  I lean towards 0. for two reeasons at the moment:
- the TICA cubes had the dummy values sent to 0. I think, so this maintains that part of it
- adding NaN may result in issues for non-NaN-safe calculations, e.g., produce values of NaN for computation that included the flux uncertainty.  Of course 0. could also introduce problems depending on what the user is doing with it, it's not obious to me which is better if someone isn't aware that these are essentially undefined but use the flux uncertainties in calcuations within their code.

Reply:
in that case wouldn't it be better to nullify the results than to silently make the calculation with an incorrect value for the error?

Reply:
that was my old philosophy: use NaN's as a forced flag to indicate to an unaware user that something is not defined, and you either need to mask or add nan-safe options to your function calls.  these days I wonder how many functions filter out NaN values automatically or not.  Do you have thoughts on this @hmlewis-astro @jaymedina ?

Reply:
+1 on the NaN's.  I just ask we make sure they are filled in such that they are read in correctly with standard astropy fits reading functions, e.g., we don't want to somehow fill them with NaN's and then have them treated as strings when default astropy functions are called on the fits products.  It's probably as easy as making sure the dtype for the FITS table column is correct and you feed them proper np.nan values I guess, although there's a small part of me that worries that having NaN's as not-strings violates a FITS table rule....of course you could then say "we'll just fill that column as strings that say NaN" but then you'll have different dtypes for TICA vs. SPOC in that column...

Reply:
Now that Ben mentions this, I'm also thinking people might assume the 0s are not fillers, especially since in the metadata we explicitly put the unit of the error array as being in electrons ([see here](https://github.com/spacetelescope/astrocut/blob/49dee429c4226de58c74d889915b454e0a17a9ab/astrocut/cube_cut.py#L731)). What we could do is instead fill the array with some ridiculously large number like 9999999 so that users know straight away that this arrays for TICA are filler arrays? Just a thought. Otherwise, I'm leaning towards making them NaN arrays now.

Reply:
@jaymedina If we go that route (non-zero but valid number) I would vote for a rediculously large NEGATIVE number (-9999999.0 or something), since negative flux uncertainties would also be treated as suspcious, but in theory you could have crazy large positive values for uncertainties.

Reply:
I'm in favor of that. I'll go ahead and make a commit unless there are other suggestions!

Reply:
I am not in favor of using any actual number here -- the error/uncertainty is unknown. using actual numbers is likely just going to result in confusion. if we're returning an array to conform with some expectation that the cutout has a certain number of extensions or whatever, but the values are not to be used -- well, this is precisely what np.NaN is for. we might also consider writing something in the header of the fits file about it?

Reply:
I'm not opposed to changing the vals to `np.nan`s. One thing to consider tho, is that `np.zero` arrays are used for all the other columns which we don't have information about, for example `CADENCEN0` which is the column that was referenced in the start of this thread (when making references to `cadence_array`). So if 1 column uses `np.nan` to fill in values we don't know about, and 1 column uses `np.zero`, that will be even more confusing.

What we can do for now is stick with the traditional `np.zero` array (I can reassign `flux_err_array` to `np.zeros(img_cube.shape)` to ensure its a zero array as suggested in the first comment of the thread), and once we come to can agreement on what the fill vals should be, I'll open up a new PR to update the `empty_arr` for all relevant columns.

As for timeline, I'd like to get this PR merged in by end of this week or beginning of next week so I have enough time to update and redeploy the TESSCut API before the end of this sprint.

Reply:
I think this solution works for now! Once you make the changes to reassign `flux_err_array` following Ben's original comment, I think this is ready to merge

Reply:
Sounds good, thanks @hmlewis-astro !